sdcclient/_client.py

```diff
@@ -111,57 +111,80 @@ def get_n_connected_agents(self):
         data = res.json()
         return [True, data['total']]
 
-    def get_notification_ids(self, channels):
-            res = requests.get(self.url + '/api/notificationChannels', headers=self.hdrs, verify=self.ssl_verify)
+    def get_notification_ids(self, channels=None):
+        '''**Description**
+            Get an array of all configured Notification Channel IDs, or a filtered subset of them.
 
-            if not self._checkResponse(res):
-                return [False, self.lasterr]
+        **Arguments**
+            - **channels**: an optional array of dictionaries to limit the set of Notification Channel IDs returned. If not specified, IDs for all configured Notification Channels are returned. Each dictionary contains a ``type`` field that can be one of the available types of Notification Channel (``EMAIL``, ``SNS``, ``PAGER_DUTY``, ``SLACK``, ``OPSGENIE``, ``WEBHOOK``) as well as additional elements specific to each channel type.
+
+        **Success Return Value**
+            An array of Notification Channel IDs (integers).
+
+        **Examples**
+            - `examples/create_alert.py <https://github.com/draios/python-sdc-client/blob/master/examples/create_alert.py>`_
+            - `examples/restore_alerts.py <https://github.com/draios/python-sdc-client/blob/master/examples/restore_alerts.py>`_
+        '''
+
+        res = requests.get(self.url + '/api/notificationChannels', headers=self.hdrs, verify=self.ssl_verify)
 
-            # Should try and improve this M * N lookup
-            ids = []
-            for c in channels:
-                found = False
-                for ch in res.json()["notificationChannels"]:
-                    if c['type'] == ch['type']:
-                        if c['type'] == 'SNS':
-                            opt = ch['options']
-                            if set(opt['snsTopicARNs']) == set(c['snsTopicARNs']):
+        if not self._checkResponse(res):
+            return [False, self.lasterr]
+
+        ids = []
+
+        # If no array of channel types/names was provided to filter by,
+        # just return them all.
+        if channels is None:
+            for ch in res.json()["notificationChannels"]:
+                ids.append(ch['id'])
+            return [True, ids]
+
+        # Return the filtered set of channels based on the provided types/names array.
+        # Should try and improve this M * N lookup
+        for c in channels:
+            found = False
+            for ch in res.json()["notificationChannels"]:
+                if c['type'] == ch['type']:
+                    if c['type'] == 'SNS':
```

Comment:
Though I don't think it shouldn't be part of this review but VictorOps is missing from this list of types. It can be compared the same way as OpsGenie, by `name`.